src/client/Inode.cc

```diff
@@ -80,8 +80,8 @@ ostream& operator<<(ostream &out, const Inode &in)
   if (in.is_file())
     out << " " << in.oset;
 
-  if (!in.dentries.empty())
-    out << " parents=" << in.dentries;
+  if (in.is_dir() && !in.dentries.empty())
```

Comment:
Seems reasonable, but, have you checked if the callers already do this check before printing out the inode?

Reply:
BTW, the `in.is_dir()` is redundant since in.dentries will always be empty for a non-directory.

Also, "parent" is the convention that's followed in the codebase, so, I would like to keep it as-it-is.

Reply:
I suppose just checking whether in.dentries is empty is not sufficient as seen at other places in ceph code. So there is need to have this check
`[psachin@rhs-srv-03 client]$ pwd
/data/ceph/ceph/src/client
[psachin@rhs-srv-03 client]$ find . | xargs grep "dentries.empty" 2>/dev/null  | grep is_dir
./Client.cc:    if (in->is_dir() && !in->dentries.empty()) {
./Client.cc:    if (in->is_dir() && !in->dentries.empty()) {
./Client.cc:    if (in->is_dir() && !in->dentries.empty()) {`

Reply:
In Client.cc file, its mentioned as dentries, as seen in below code snippet. So wondering should it be called as "parents" or "dentries"
```
 1151       if (in->dir) {
 1152         ldout(cct, 10) << " dir is open on empty dir " << in->ino << " with "
 1153                        << in->dir->dentries.size() << " entries, marking all dentries null" << dendl;
 1154         in->dir->readdir_cache.clear();
 1155         for (const auto& p : in->dir->dentries) {
 1156           unlink(p.second, true, true);  // keep dir, keep dentry
 1157         }
 1158         if (in->dir->dentries.empty())
 1159           close_dir(in->dir);
 1160       }
```

Reply:
> In Client.cc file, its mentioned as dentries, as seen in below code snippet. So wondering should it be called as "parents" or "dentries"

So, this naming is subjective - for me, using "parents" is very much what the data structure is used for.

Reply:
> I suppose just checking whether in.dentries is empty is not sufficient as seen at other places in ceph code. So there is need to have this check

OK. So I rechecked. inode->dentries is the list of parents an inode is linked to - can be a file inode or a directory.

Inode.h: xlist<Dentry *> dentries; // if i'm linked to a dentry.

So, the is_dir() check isn't really required.

Reply:
okay, will close this PR.